lib/Controller/ChatController.php

```diff
@@ -472,6 +472,11 @@ public function receiveMessages(int $lookIntoFuture,
 			$message = $this->messageParser->createMessage($this->room, $this->participant, $comment, $this->l);
 			$this->messageParser->parseMessage($message);
 
+			$expireDate = $message->getComment()->getExpireDate();
+			if ($expireDate instanceof \DateTime && $expireDate < $this->timeFactory->getDateTime()) {
```

Comment:
Calling `$this->timeFactory->getDateTime()` 200 times here. Want to move it outside of the loop?

Also should add it around line 520 where parents are rendered and look at other endpoints (search comes to mind)